compiler/noirc_frontend/src/hir/def_collector/errors.rs

```diff
@@ -100,6 +100,19 @@ impl DefCollectorErrorKind {
     }
 }
 
+impl<'a> From<&'a UnsupportedNumericGenericType> for Diagnostic {
+    fn from(error: &'a UnsupportedNumericGenericType) -> Diagnostic {
+        let name = &error.ident.0.contents;
+        let typ = &error.typ;
+
+        Diagnostic::simple_error(
+            format!("{name} has a type of {typ}. The only supported types of numeric generics are integers, fields, and booleans."),
```

Comment:
```suggestion
            format!("{name} has a type of {typ}. The only supported numeric generic types are `u1`, `u8`, `u16`, and `u32`."),
```
I see this was copied from the error that existed before. Let's just update it to match the error in the parser. I guess this hasn't been hit as the parser has been catching it, but good to be aligned here.